### PR TITLE
Additional Movie Parts

### LINKED ISSUE #830: Cannot get Extra Videos to show as options
**Describe the bug**
Extra features do not show up as video options. I think this might be a regression.

As an example I have The Fellowship of the Ring Extended Edition named:

`/mnt/Media/Movies/The Lord of the Rings The Fellowship of the Ring (2001)/The Lord of the Rings The Fellowship of the Ring (2001) - Extended Edition Part 1.m4v`
`/mnt/Media/Movies/The Lord of the Rings The Fellowship of the Ring (2001)/The Lord of the Rings The Fellowship of the Ring (2001) - Extended Edition Part 2.m4v`

But in the Roku app I can only watch part 1.

**Expected behavior**

Ideally we would be able to watch Part 2 by selecting it as an option.

### PR DESCRIPTION
**Pull request by jimdogx**
Show any "additional parts" (part 2, part 3, etc) of a movie.

**Changes**
Adds a new row of additional parts (if any are found) so that the user can continue watching Part2, Part 3, etc.

**Issues**
Fixes #830

### CODE REVIEW

**Comment by 1hitsong** [comment]
👍 Works as intended.

The only issue I see is as the final row in the extras slide up, it's pretty well hidden. Can we check if additional parts exist and add a button to the main details page that says "Play Additional Parts" and popups up a list of found parts to play?

**Comment by jimdogx** [comment]
> 👍 Works as intended.
> 
> The only issue I see is as the final row in the extras slide up, it's pretty well hidden. Can we check if additional parts exist and add a button to the main details page that says "Play Additional Parts" and popups up a list of found parts to play?

We could do that or we could just make sure additional parts is always first in the list (when additional parts are found of course).  Preference?

**Comment by 1hitsong** [comment]
Using the issue as an example, we're talking about playing the 2nd half of the movie. It just feels wrong to have the playback button for it anywhere the user can't see immediately.

**Comment by jimdogx** [comment]
> Using the issue as an example, we're talking about playing the 2nd half of the movie. It just feels wrong to have the playback button for it anywhere the user can't see immediately.

I was mostly trying to be consistent with the web client:
Screenshot 2022-11-22 at 8 32 35 AM

But there is no rule that says we need to do that (specially if we think we can do it better) 😄

**Comment by frob** [comment]
Bringing my comment from the issue into the PR:

> I tested that build and it does show a section labeled "Additional Features" though it was hard to find.

> This new section is below "Cast & Crew" and the "More like this" section. It should probably be above those other sections and labeled "Additional Parts" which is how the web client presents it.

> Another bug I noticed with this was the screenshot isn't shown in a screen shot image style but rather if displays in a portrait style like the head shots and box art.

**Comment by frob** [comment]
> We could do that or we could just make sure additional parts is always first in the list (when additional parts are found of course). Preference?

Another consideration is that they don't just say "Part 2" or "Part 3"

In the screenshot you post it is "Isle of Destiny - Part 2" in my example from the issue it would be "The Lord of the Rings The Fellowship of the Ring (2001) - Extended Edition Part 2" or really what gets show in the UI is something like "The L ... art 2"

**Comment by jimdogx** [comment]
@frob I've pushed a changed that moves "Additional Parts" to the top (if found), just like on the web client.  Let me know if this works for you.

Also, the title of the Part 2 movie should scroll if it is too long, is it not scrolling for you?

**Comment by 1hitsong** [comment]
This is better. 

Is there anything we can do with the thumbnail? Can we change it to a wide thumbnail instead of a vertical poster image?

**Comment by frob** [comment]
image

It would be nice if the image was 16:9 or at least not squished.

image

Additional Parts may still get lost with the text muted the way it is. Giving it a bit more line-height or even a tab like treatment may make it stand out more. If that is difficult then making the text white should do the trick.

**Comment by jimdogx** [comment]
@frob @1hitsong I agree the thumbnail needs adjusting.  I'll see if I can do that one of these days.

As for the "Additional Parts" being muted, that is our current way of subtly hinting that there is more stuff below.  Not saying it is perfect or won't get redesigned in the future, but if you're using the Roku client you'll already be used to this pattern for everything else (e.g. Cast and Crew, More Like This, etc are all presented this way).

**Comment by 1hitsong** [comment]
Perhaps in a follow up PR we can look into using some of the slider logic I made for the artist view.

File: components/extras/ExtrasRowList.brs [hunk 1]
```diff
@@ -41,12 +45,30 @@ sub loadPersonVideos(personId)
...
+
+    data = CreateObject("roSGNode", "ContentNode") ' The row Node
+    m.top.content = data
+    if parts <> invalid and parts.count() > 0
+        row = buildRow("Additional Parts", parts)
+        addRowSize([234, 396])
+        m.top.content.appendChild(row)
+    end if
+    m.top.translation = "[75,10]"
+    m.top.rowItemSize = [[234, 396]]
```
Comment (1hitsong) [inline components/extras/ExtrasRowList.brs]:
```suggestion
        row = buildRow("Additional Parts", parts, 464)
        addRowSize([464, 291])
        m.top.content.appendChild(row)
    end if
    m.top.translation = "[75,10]"
    m.top.rowItemSize = [[464, 291]]
```

**Review by 1hitsong — CHANGES_REQUESTED** [review]

File: components/extras/ExtrasRowList.brs (hunk 1 above)
Comment (1hitsong) [reply components/extras/ExtrasRowList.brs]: I believe this will make the thumbnail image be horizontal.

Comment (jimdogx) [reply components/extras/ExtrasRowList.brs]: Unfortunately that makes the first row wide no matter what the first row is (including Cast and Crew).  I'm taking a look now.

**Comment by jimdogx** [comment]
@frob @1hitsong I think it's finally working correctly.  Wide thumbnail for Additional Parts, but keeping the slimmer aspect for Cast and Crew etc.

**Review by 1hitsong — APPROVED** [review]

**Comment by 1hitsong** [comment]
@frob Found a bug in this that can cause the user to get stuck with no way out other than turning the roku off/on.

I've notified @jimdogx about the underlying issue, but the fix isn't very straightforward.

**Pull request merged**